spec.bs

```diff
@@ -1226,6 +1254,280 @@ On the other hand, methods for getting data from the [=shared storage database=]
+
+  The {{RequestInit}} dictionary contains a {{RequestInit/sharedStorageWritable}} key:
+
+<xmp class='idl'>
+  partial dictionary RequestInit {
+    boolean sharedStorageWritable;
+  };
+</xmp>
+
+## Fetch Algorithm Modifications ## {#mod-fetch-algo}
+
+### Modification to the [=request-constructor|Request Constructor=] Algorithm ### {#mod-request-con}
+
+  <div algorithm="modification-to-request-constructor">
+    Modify the [=request-constructor|new Request(input, init) constructor=] as follows:
+
+    Before the step
+
+    > Set this's [=Request/request=] to |request|.
+
+    add the step
+
+    1. If <var ignore=''>init</var>["{{RequestInit/sharedStorageWritable}}"] [=map/exists=], then set |request|'s [=request/shared storage writable=] to it.
+  </div>
+
+### Modification to [=HTTP network or cache fetch=] Algorithm ### {#mod-http-net-fetch}
+
+  <div algorithm="modification-to-http-network-or-cache-fetch">
+    Modify the [=HTTP network or cache fetch=] algorithm as follows:
+
+    Before the step
+
+    > Modify |httpRequest|'s [=request/header list=] per HTTP. ...
+
+    add the step
+
+    1. [=Append or modify a Shared-Storage-Writable request header=] for |httpRequest|.
+  </div>
+
+### Modification to [=HTTP fetch=] Algorithm ### {#mod-http-fetch}
+
+  <div algorithm="modification-to-http-fetch">
+    Modify the [=HTTP fetch=] algorithm as follows:
+
+    Before the step
+
+    > If |internalResponse|'s [=response/status=] is a [=redirect status=]: ...
+
+    add the step
+
+    1. [=Handle a Shared-Storage-Write response=], given [=/response=] |internalResponse| and [=/request=] <var ignore=''>request</var> as input.
+  </div>
+
+## Shared Storage HTTP Headers ## {#headers}
+
+### [=Shared-Storage-Writable=] Request Header ### {#request-header}
+
+  This specification defines a <dfn>Shared-Storage-Writable</dfn> HTTP [=/request=] [=header=].
+
+  The [=Shared-Storage-Writable=] [=/request=] [=header=] is a [=item/boolean|boolean structured header=].
+
+  When the result of running [=get a structured field value=] for "[=Shared-Storage-Writable=]", "`item`", and [=/request=]'s [=request/header list=] is true, then the [=/response=] to this [=/request=] will be eligible to write to [=shared storage=].
+
+### [=Shared-Storage-Write=] Response Header ### {#response-header}
+
+  This specification defines a <dfn>Shared-Storage-Write</dfn> HTTP [=/response=] [=header=].
+
+  The [=Shared-Storage-Write=] [=/response=] [=header=] is a [=list structured header=], for which the [=parameters|parameterized=] [=structured header/items=] can have one of the following values, expressed as [=item/token=], [=item/string=], or [=item/byte sequence=]:
+
+    * `set`
+        - required [=parameters=]:
+            * key ([=item/Token=], [=item/String=], or [=item/Byte Sequence=])
+            * value ([=item/Token=], [=item/String=], or [=item/Byte Sequence=])
```

Comment:
This could use a note about why it's important to allow byte sequences in these values. Is it to allow unicode? Also specify that they're interpreted as UTF-8.

Reply:
Added